docs/source/basics/messages.rst

```diff
@@ -0,0 +1,144 @@
+.. _messages_usage_label:
+
+========
+Messages
+========
+
+Messages are a core component in torchtune that govern how text and multimodal content is tokenized. It serves as the common interface
+for all tokenizer and datasets APIs to operate on. Messages contain information about the text content, which role is sending the text
+content, and other information relevant for special tokens in model tokenizers. For more information about the individual parameters
+for Messages, see the API ref for :class:`~torchtune.data.Message`. Here, we briefly discuss how to create messages, format messages, and
+tokenize messages.
+
+.. _creating_messages:
+
+Creating Messages
+-----------------
+
+Messages can be created via the standard class constructor or directly from a dictionary.
+
+.. code-block:: python
+
+    from torchtune.data import Message
+
+    msg = Message(
+        role="user",
+        content="Hello world!",
+        masked=True,
+        eot=True,
+        ipython=False,
+    )
+    # This is identical
+    msg = Message.from_dict(
+        {
+            "role": "user",
+            "content": "Hello world!",
+            "masked": True,
+            "eot": True,
+            "ipython": False,
+        },
+    )
+    print(msg.content)
+    # [{'type': 'text', 'content': 'Hello world!'}]
+
+Content is formatted as a list of dictionaries. This is because Messages can also contain multimodal content, such as images.
+
+Images in Messages
+^^^^^^^^^^^^^^^^^^
```

Comment:
This code is great, but i think we need a bit of end2end, showing how to create the dataset and where messages are used.

Reply:
was actually thinking of a section "Tokenizing Messages" or similar to show how they are consumed